xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -810,6 +813,16 @@ void CGUIWindowManager::CloseDialogs(bool forceClose) const
   }
 }
 
+void CGUIWindowManager::CloseModalDialogs(bool forceClose) const
+{
+  CSingleLock lock(g_graphicsContext);
```

Comment:
After a second look every dialog related method (e.g. RemoveDialog, HasModalDialog etc.) use a graphical lock.
Changing this is out of scope of this PR IMO.